client/src/partials/vouchers/complex.html

```diff
@@ -101,164 +101,49 @@
-                      ng-click="ComplexVoucherCtrl.openReferenceModal(row)"
-                      ng-class="{'btn-info': row.reference.uuid}"
-                      data-reference-button>
-
-                      <!-- state: has reference -->
-                      <span ng-show="row.reference.uuid">
-                        <i class="fa fa-file-o"></i> {{ row.reference.reference }}
-                      </span>
-
-                      <!-- state: no reference -->
-                      <span ng-hide="row.reference.uuid">
-                        <i class="fa fa-plus"></i> {{ 'VOUCHERS.COMPLEX.REFERENCE' | translate }}
-                      </span>
-                    </button>
-                  </td>
-                  <td class="text-right">
-                    <button
-                      class="btn btn-danger"
-                      type="button"
-                      ng-click="ComplexVoucherCtrl.removeVoucherItem($index)">
-                      <i class="fa fa-trash-o"></i>
-                    </button>
-                  </td>
-                </tr>
-              </tbody>
-              <tfoot>
-                <tr>
-                  <th colspan="3">{{ ComplexVoucherCtrl.rows.length | number }} {{ 'VOUCHERS.COMPLEX.ROWS' | translate }}</th>
-                  <th class="text-right">{{ ComplexVoucherCtrl.sumDebit | currency:ComplexVoucherCtrl.voucher.currency_id }}</th>
-                  <th class="text-right">{{ ComplexVoucherCtrl.sumCredit | currency:ComplexVoucherCtrl.voucher.currency_id }}</th>
-                  <th colspan="3">&nbsp;</th>
-                </tr>
-              </tfoot>
-            </table>
-
-            <div class="panel-footer clearfix">
+          <!-- the heading  -->
+          <div class="space-between" style="margin-bottom: 5px;">
+            <!-- Hack styles to make text behave look like class "btn".  This should be formalized into a class -->
+            <span style="display: inline-block; padding: 6px 12px; vertical-align:middle;">
+              <i class="fa fa-list"></i> {{ 'FORM.LABELS.TRANSACTION' | translate }}s
+            </span>
+
+            <button
+              type="button"
+              class="btn btn-default pull-right"
+              ng-click="ComplexVoucherCtrl.addVoucherItem()"
+              data-button-add-item>
+              <i class="fa fa-plus-circle"></i> {{ 'VOUCHERS.COMPLEX.ADD_ITEMS' | translate }}
+            </button>
+          </div>
 
-              <span ng-class="{'text-danger': !ComplexVoucherCtrl.validInput, 'text-success': ComplexVoucherCtrl.validInput}" ng-hide="ComplexVoucherCtrl.posted">
-                <span ng-class="ComplexVoucherCtrl.notifyMessage.icon"></span>
-                {{ ComplexVoucherCtrl.notifyMessage.label | translate }}
-              </span>
+          <!-- the grid  -->
+          <div id="voucherGridId"
+            ui-grid="ComplexVoucherCtrl.gridOptions"
+            ui-grid-auto-resize
+            ui-grid-resize-columns>
```

Comment:
I don't think resizing columns is a good idea on a grid with inputs.  What do you think?  Is it really necessary?